homeassistant/components/shelly/config_flow.py

```diff
@@ -83,7 +83,7 @@ async def async_step_user(self, user_input=None):
                     errors["base"] = "unknown"
                 else:
                     return self.async_create_entry(
-                        title=device_info["title"] or self.host,
+                        title=device_info["title"] or device_info["mac"],
```

Comment:
We should use the hostname, we've been using that for name too.

Reply:
Does `device.settings` contain the hostname? If not, I'm not sure how to get it for the non-discovery use case.

Reply:
@balloob ok, I can get `device.settings["device"]["hostname"]`. The problem is that the type of device is taking up all of the screen space that way:
<img width="299" alt="Screen Shot 2020-10-01 at 11 15 25 AM" src="https://user-images.githubusercontent.com/2159693/94784868-77576400-03d7-11eb-8a00-8c4527cfab1c.png">

Reply:
hmm maybe we can remove `shelly` from the beginning, but still have `switch25` there to help identifying it? (also, we should update the name cut off in the UI, but that's a different issue)

Reply:
OK, tried that - works well for the 2.5, but looks weird on Shelly 1, which is now just `1-<mac address>`. I really think just mac is the best approach.

Reply:
Also, while not perfect, I think mac is an improvement over ip, and if we change the UI to accommodate larger titles in the future, we can switch to device id.

Reply:
Let's stick to hostname because we use that for name too. We should not work around frontend issues in our backend.

Reply:
In principal I agree with not working around frontend issues in the backend, but until the issue is resolved, hostname might be worse than IP since all the text that differentiates the devices will be hidden.

I guess the best suggestion is to table this until we update the UI.

Reply:
@balloob another idea: what if we remove the "Shelly: " prefix from the flow title? So:
```json
"flow_title": "Shelly: {name}",
```
becomes:
```json
"flow_title": "{name}",
```

I think that ideally we should also use Shelly as an alt text for the logo image for accessibility, but the hostname start with shelly anyway.

Reply:
Great idea!

Reply:
OK, I'm now using the hostname, and dropped the "Shelly: " prefix from the flow title. Additionally, if the hostname starts with "shellyswitch", I also drop the "shelly" prefix from it.